sparse/impl/KokkosSparse_trsv_impl.hpp

```diff
@@ -261,10 +261,8 @@ struct TrsvWrap {
       return;
     }
 
-    // Don't use r >= 0 as the test, because that fails if
-    // lno_t is unsigned.  We do r == 0 (last
-    // iteration) below.
-    for (lno_t r = numRows - 1; r != 0; --r) {
+    // Iterate backwards with care due to potentially unsigned type
+    for (lno_t r = numRows; r-- > 0;) {
```

Comment:
I am not a big fan of `r-- > 0` because it is a bit harder to understand but does not really make the code much smaller or more performant. Also is the loop now starting at `numRows` instead of `numRows - 1`?

Reply:
> Also is the loop now starting at numRows instead of numRows - 1?

Yes, because `r` gets `--`'d before it gets used in the loop body.

Reply:
So what would be the issue with just rewriting this as: `for (lno_t r = numRows; r > 0; --r)` and then just index with `r - 1`?
Does it lead to the same issue with the case `r == 0`

Reply:
@lucbv , I think that would work too but you'd need to change all the indexes. Did you see my static_cast option below?